lib/ansible/modules/system/iocage.py

```diff
@@ -0,0 +1,615 @@
+  iocage:
+    name: test
+    release: 12.0-RELEASE
+    
+- name: > 
+      ensure jail is created with a certain 
+      zfs pool activated
+  iocage:
+    zpool: zroot
+    name: test
+    release: 12.0-RELEASE
+
+- name: Ensure jail is started
+  iocage:
+    name: test
+    release: 12.0-RELEASE
+    state: started
+
+
+- name: Ensure jail is stopped
+  iocage:
+    name: test
+    release: 12.0-RELEASE
+    state: present
+
+- name: Ensure jail is destroyed
+  iocage:
+    name: test
+    state: absent
+    
+- name: ensure jail properties are set
+  iocage:
+    name: test
+    state: present
+    release: 12.0-RELEASE
+    properties:
+      boot: "on"
+      vnet: "on"
+      ip4_addr: "192.168.1.2/24"
+      defaultrouter: "192.168.1.1"
+      '''
+
+RETURN = '''
+changed: 
+    description: indicator if ansible has changed a jail
+    type: bool
+    returned: always    
+failed: 
+    description: indicator if everything went ok
+    type: bool
+    returned: always
+message: 
+    description: what steps have been changed
+    type: str
+    returned: always but empty string if nothing has been done
+name:
+    description: the jail's name
+    type: str
+    returned: if the jail has got a name
+properties: 
+    description: >
+        all the jail's properties, see iocage(8) for
+        explanations about their meanings 
+    type: dict
+    returned: if the jail exists after a successful run
+release: 
+    description: the FreeBSD release the jail is based upon
+    type: str
+    returned: if the jail exists after a successful run
+state:
+    description: the jail's state, see I(state)
+    type: str
+    returned: if the jail exists after a successful run
+uuid:
+    description: the jail's uuid
+    type: str
+    returned: if the jail exists after a successful run
+zpool:
+    description: the zpool activated for iocage
+    type: str
+    returned: if the jail exists after a successful run'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import to_text
+from typing import Dict
+from typing import List
+
+
+class Jail:
+
+    def __init__(self, name: str, uuid: str = None, release: str = None, template: str = None,
+                 empty: bool = False, state: str = False,
+                 properties: Dict = None):
+        self.name = name
+        self.uuid = uuid
+        self.release = release
+        self.template = template
+        self.empty = empty
+        self.state = state
+        self.properties = properties
+
+
+class IOCage:
+    IOCAGE = ["iocage"]
+
+    def __init__(self, module: AnsibleModule, result: Dict, zpool: str = None):
+        self.module = module
+        self.result = result
+        self.zpool = zpool
+
+    def is_activated(self, zpool: str) -> bool:
+        """
+        Check if the given zpool has been activated for use with iocage
+        """
+        stdout = self.get_activated_zpool()
+        if stdout.strip() == zpool:
```

Comment:
`stdout.strip()` is done twice (`get_activated_zpool()` already does it).

```suggestion
        if stdout == zpool:
```